CRM/Core/EntityTokens.php

```diff
@@ -120,18 +120,12 @@ public function evaluateToken(TokenRow $row, $entity, $field, $prefetch = NULL)
       return $row->customToken($entity, \CRM_Core_BAO_CustomField::getKeyID($field), $this->getFieldValue($row, 'id'));
     }
     if ($this->isMoneyField($field)) {
-      $currency = $this->getCurrency($row);
+      $currency = $this->getCurrency($row) ?: \Civi::settings()->get('defaultCurrency');
       if (empty($fieldValue) && !is_numeric($fieldValue)) {
         $fieldValue = 0;
       }
-      if (!$currency) {
```

Comment:
this basically just uses the defaultCurrency - so the above is correct & less flakey